File: simple_numbers_10^6.cpp
#include <iostream>
#include <math.h>

using namespace std;

int main() {
	
	int m, n;
	cin >> m >> n;
	
	n++;

	bool primes[n];
	primes[0] = primes[1] = false;
	for (int i = 2; i < n; i++) primes[i] = 1;
	
	int sqrtN = sqrt(n) + 1;
	
	for (int i = 0; i < sqrtN; i++) {
		if (primes[i]) {
			for (int j = i*i; j < n; j += i) {
				primes[j] = false;
			}
		}
	}
	
	bool wasSimple = 0;
	
	for (int i = m; i < n; i++) {
		if (primes[i]) {
			wasSimple = 1;
			cout << i << endl;
		}
	}

	if (!wasSimple)
		cout << "Absent";
	

}

File: simple_numbers_10^6.py
from time import time

m, n = map(int, input().split())

start = time()

n += 1
primes = [True] * n
primes[0] = primes[1] = False

sqrtN = int(n ** 0.5) + 1

for i in range(2, sqrtN):
    if primes[i]:
        primes[i] = True

        for j in range(i*i, n, i):
            primes[j] = False

wasSimple = False



#for i in range(m, n):

#    if primes[i]:
#        wasSimple = True
#        print(i)

#if not wasSimple:
#    print("Absent")


end = time()


print(end - start)



